### ISSUE TITLE
Python interface on OS X 

### ISSUE THREAD

**Issue by teancake** [bgrimstad/splinter#69]
Hi, on my macbook (OS X El Captain V10.11.6, Python 2.7.11, Apple LLVM version 7.3.0 (clang-703.0.31), Target: x86_64-apple-darwin15.6.0), I got the following error: 
```
In [1]: import splinter
Unable to automatically locate SPLINTER.
You can load it manually by doing splinter.load("/path/to/SPLINTER.so")

```
This error happened both in the [released version](https://github.com/bgrimstad/splinter/releases/download/v3.0/splinter-python.tar.gz) and my compiled version. My compilation worked successfully with two lib files 'libsplinter-3-0.dylib' and 'libsplinter-static-3-0.a' generated, but it seemed that the 'libsplinter-x-y.so' file described in the [python interface](https://github.com/bgrimstad/splinter/blob/master/docs/python_interface.md) was missing. 

I am not sure whether this is due to the configuration of my computer or not, could anyone kindly help to resolve this problem? 

Thanks.

**Comment by gablank** [comment]
Hello @teancake,

We do not have access to any machines running OS X, so we have actually have been unable to test the library sufficiently on that platform.

The problem with the precompiled / released version is that there are no precompiled binaries available for OS X (if we haven't written that in the release I'll be sure to add it), so the Python side is unable to find any files (because they don't exist).

As you've noticed, we are using CMake for generating the build files. It seems that the suffix of the shared library is not what I thought it would be (.so), so the Python interface therefore is unable to find the file, again. You could try to split [this](https://github.com/bgrimstad/splinter/blob/master/python/splinter/splinter.py#L206) statement into two, one for Linux and one for OS X, and then change the filename of the OS X part to have .dylib as suffix instead of .so.

An easy way to test your fix before implementing it would be to do this:
```python
import splinter
# Expect error printout here, because SPLINTER was unable to auto-load the library
splinter.load("path/to/splinter/libsplinter-3-0.dylib")
```

Also keep in mind that the architecture (x86 vs x86-64) for your Python and SPLINTER binaries must match. I am unsure of how it works on OS X (as I believe they have some dual architecture binaries?).

If you have any further questions or feedback, we will be happy to assist you or fix any bugs you might encounter!

Sincerely,
Anders.

**Comment by teancake** [comment]
Thanks @gablank ,  by splitting the [if statement](https://github.com/bgrimstad/splinter/blob/master/python/splinter/splinter.py#L206) as 
```python
    elif is_linux: 
        splinter_name = "lib" + splinter_basename + ".so"
    elif is_mac:
        splinter_name = "lib" + splinter_basename + ".dylib"
```
I could successfully load splinter. 

Thanks very much.
Xiaoke

**Comment by gablank** [comment]
Good to hear it worked, and thanks for letting us know so we can implement the fix!

**Comment by gablank** [comment]
Fixed by 480d69060f65490cdd5e886bcca00f1090ff7691

**Comment by christovski** [comment]
Can confirm that this issue was still in the release 3.0 version, but could be patched by changing '.so' -> to '.dylib' on line 55 of the setup() function.
